### PR TITLE
Moved retry (Polly) code into extensions methods

### PR DESCRIPTION
**Pull request by mvanbeusekom**

### CODE REVIEW

File: src/Xablu.WebApiClient/BaseClient.cs [hunk 1]
```diff
@@ -19,29 +18,7 @@ public BaseClient(IWebApiClient apiClient)
...
-            {
-                if (e.GetType().Namespace == "Java.Net")
-                {
-                    throw new HttpExtensions.HttpResponseException(HttpStatusCode.RequestTimeout, e.Message, e.StackTrace);
-                }
-                throw;
-            }
-
-            return result;
+            return await WebApiClientPollyExtensions.PollyDecorator(action, 5, 5);
```
Comment (martijn00) [inline src/Xablu.WebApiClient/BaseClient.cs]: I don't think you need to await this, but just return the Task.

Comment (martijn00) [inline src/Xablu.WebApiClient/BaseClient.cs]: Maybe a `sleepDuration` of 5 seconds is a bit long for default? Is it an idea to change this to 1 or 2? And maybe the retry to 3?

File: src/Xablu.WebApiClient/Xablu.WebApiClient.csproj [hunk 2]
```diff
@@ -92,9 +93,7 @@
   <ItemGroup>
     <None Include="packages.config" />
   </ItemGroup>
-  <ItemGroup>
```
Comment (MarcBruins) [inline src/Xablu.WebApiClient/Xablu.WebApiClient.csproj]: Is this not removing the ProgressStreamContent.cs from the project?

Comment (mvanbeusekom) [inline src/Xablu.WebApiClient/Xablu.WebApiClient.csproj]: No it won't it only removes the explicit folder declaration. The folder will be implicitly inferred from the file declaration on line 52: `<Compile Include="HttpContentExtensions\ProgressStreamContent.cs" />`

So the file and even the folder will still be there.

Comment (MarcBruins) [inline src/Xablu.WebApiClient/Xablu.WebApiClient.csproj]: Ah didn't know that! That's cool, i'll remember that 👍

File: src/Xablu.WebApiClient/BaseClient.cs (hunk 1 above)
Comment (mvanbeusekom) [inline src/Xablu.WebApiClient/BaseClient.cs]: I also think 5 is a bit long. In the previous version this was actually increased in an interval (`TimeSpan.FromSeconds(Math.Pow(2, retryAttempt))`). But this implies you'll need to supply a function to the method which I left out to keep it simple. Maybe I can add another overload so you could pick between simply waiting for some seconds or you can supply a function to make a more complicated calculation.

Regarding the other comment, I think you are right and will simply return the task.

Comment (mvanbeusekom) [inline src/Xablu.WebApiClient/BaseClient.cs]: Added support for more complex sleep strategies and updated the code in the `BaseClient` class to reflect the original situation (where sleep duration is increased by the power of 2 each retry attempt)

**Pull request merged**